app/main.py

```diff
@@ -2,0 +1,43 @@
+class Animal:
+    alive = []
+
+    def __init__(
+            self,
+            name: str,
+            health: int = 100,
+            hidden: bool = False
+    ) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = hidden
+        Animal.alive.append(self)
+
+    def __del__(self) -> None:
+        if self.hidden and self in Animal.alive:
+            Animal.alive.remove(self)
+
+    def hide(self) -> None:
+        self.hidden = not self.hidden
+
+    def __str__(self) -> str:
+        return f"{{Name: {self.name}, Health: {self.health}, Hidden: {self.hidden}}}"
+
+    def __repr__(self) -> str:
+        return str(self)
+
+
+class Herbivore(Animal):
+
+    def hide(self) -> None:
+        super().hide()
```

Comment:
The `hide` method in `Herbivore` simply calls the base class method without adding any new functionality. If no additional behavior is needed, you might not need to override this method.